app/helpers/exercise_input_helper.rb

```diff
@@ -47,15 +47,34 @@ def should_render_need_help_dropdown?(assignment, organization = Organization.cu
     !assignment.passed? && organization.ask_for_help_enabled?
   end
 
-  def render_submit_button(exercise)
+  def render_submit_button(exercise, assignment=nil)
     options = submit_button_options(exercise)
     text = t(options.t) if options.t.present?
     waiting_text = t(options.waiting_t) if options.waiting_t.present?
-    %Q{<#{options.tag} for="#{options.for}"
+    %Q{
+      <div class="btn-submit-container">
+        <#{options.tag} for="#{options.for}"
                        class="btn btn-success btn-block btn-submit #{options.classes}"
                        data-waiting="#{waiting_text}">
-          #{fa_icon options.fa_icon, text: text}
-       </#{options.tag}>}.html_safe
+          #{fa_icon options.fa_icon}
+          #{text} #{remaining_attempts_text(exercise, assignment)}
+       </#{options.tag}>
+      </div>
+    }.html_safe
+  end
+
+  def should_disable_button?(exercise, assignment)
```

Comment:
Passing assignment and exercise is redundant - if you have the assignment, you have the exercise.

Reply:
Yes, you do. But the first time this view is rendered no assignment is created yet. And i thought it was better to take advantage of the assignment that was set by the controller, instead of calling that method again in the view.